docs/architecture/workflow-update.md

```diff
@@ -187,42 +225,65 @@ API call is returned. Currently, it can only be `ACCEPTED` or `COMPLETED`.
 > also allow using an Update as a "fire-and-forget" (just like Signal).
 
 ### Waiters
+Server performs few checks before blocking on corresponding future:  
+```mermaid
+flowchart TD
+    updateWorkflowExecution[UpdateWorkflowExecution / PollWorkflowExecutionUpdate] --> wfExists{Workflow exists?}
+    wfExists --> |yes| wfRunning{Workflow running?}
+    wfExists --> |no| notFound(((NotFound)))
+    wfRunning --> |no| updateExists{Update exists?}
+    wfRunning --> |yes| waitFor{Wait stage}
+    updateExists --> |yes| success1(((Update result)))
+    updateExists --> |no| wfCompleted(((ErrWorkflowCompleted)))
+
+    waitFor --> |ACCEPTED| blockAccepted[block on `accepted` future]
+    waitFor --> |COMPLETED| blockCompleted[block on `completed` future]
```

Comment:
Some of this isn't rendering correctly

Reply:
![image](https://github.com/user-attachments/assets/662c5c14-b6a9-4c36-b22f-bcabd1c1bd2f)